### PR TITLE
Add unittests for task1 functionalities

### PR DESCRIPTION
**Pull request by mnjaggah**
### Unittests  for:
- print_room(roomname)
- print_allocations(filename)
- print_unallocated(filename)

### CODE REVIEW

File: test_cases/test_dojo.py [hunk 1]
```diff
@@ -43,4 +45,34 @@ def test_create_room_returns_error_message_if_roomname_not_string(self):
...
+	def test_if_prints_allocated_filepath_exists(self):
+		self.the_dojo.print_allocations("file1")
+		self.assertTrue(os.path.isfile("file1.txt"))
+		os.remove("file1.txt")
+
+	def test_it_print_unallocated_filepath_exists(self):
+		self.the_dojo.print_unallocated("file2")
+		self.assertTrue(os.path.isfile("file2.txt"))
+		os.remove("file2.txt")
+
```
Comment (GDP254) [inline test_cases/test_dojo.py]: Looks good to me.

Comment (jomomg) [inline test_cases/test_dojo.py]: I see no issues

File: test_cases/test_dojo.py [hunk 2]
```diff
@@ -43,4 +45,34 @@ def test_create_room_returns_error_message_if_roomname_not_string(self):
 		with self.assertRaises(ValueError):
 			self.the_dojo.create_room("OFFICE",123)
 	
+	def test_that_a_room_exists_before_printing_occupants(self):
+		self.the_dojo.create_room("office","Green")
+		self.assertTrue(self.the_dojo.print_room("Green"))
```
Comment (Gchesta) [inline test_cases/test_dojo.py]: Great idea to test this. However, the print function returns `None` unless you specifically command it to return a value. To carry out a unit test on you must "capture"  the output and then compare it with the expected output (in this case "Green")

File: test_cases/test_dojo.py [hunk 3]
```diff
@@ -43,4 +45,34 @@ def test_create_room_returns_error_message_if_roomname_not_string(self):
...
+
+	def test_print_room_returns_error_message_if_room_doesnot_exist(self):
+		self.the_dojo.create_room("office","blue")
+		self.assertEqual(self.the_dojo.print_room("red"), "This room doesnot exist!")
```
Comment (Gchesta) [inline test_cases/test_dojo.py]: spelling "does not". Can give you a false error

File: test_cases/test_dojo.py [hunk 4]
```diff
@@ -43,4 +45,34 @@ def test_create_room_returns_error_message_if_roomname_not_string(self):
...
+
+	def test_print_room_prints_all_occupants_if_room_is_office(self):
+		self.the_dojo.create_room("office","Purple")
+		self.the_dojo.add_person("Pete","pete@pete","Staff")
+		self.assertEqual(self.the_dojo.print_room("Purple"), "Pete")
```
Comment (Gchesta) [inline test_cases/test_dojo.py]: Same as the first comment

File: test_cases/test_dojo.py [hunk 5]
```diff
@@ -43,4 +45,34 @@ def test_create_room_returns_error_message_if_roomname_not_string(self):
...
+
+	def test_print_room_prints_all_occupants_if_room_is_livingspace(self):
+		self.the_dojo.create_room("livingspace", "Ruby")
+		self.the_dojo.add_person("Marie", "marie@marie.com","Fellow", "y")
+		self.assertEqual(self.the_dojo.print_room("Ruby"), "Marie")
```
Comment (Gchesta) [inline test_cases/test_dojo.py]: Same as the first comment

File: test_cases/test_dojo.py [hunk 6]
```diff
@@ -43,4 +45,34 @@ def test_create_room_returns_error_message_if_roomname_not_string(self):
...
+
+	def test_if_prints_allocated_filepath_exists(self):
+		self.the_dojo.print_allocations("file1")
+		self.assertTrue(os.path.isfile("file1.txt"))
+		os.remove("file1.txt")
+
+	def test_it_print_unallocated_filepath_exists(self):
+		self.the_dojo.print_unallocated("file2")
+		self.assertTrue(os.path.isfile("file2.txt"))
+		os.remove("file2.txt")
```
Comment (Gchesta) [inline test_cases/test_dojo.py]: These will test whether the file was created. How will you know whether the content was actually written in there?

File: test_cases/test_dojo.py (hunk 2 above)
Comment (mnjaggah) [reply test_cases/test_dojo.py]: if i'm getting you correctly, an assertEqual() would be a better test?

Comment (GDP254) [reply test_cases/test_dojo.py]: I think that is what he means

File: test_cases/test_dojo.py [hunk 7]
```diff
@@ -43,4 +45,76 @@ def test_create_room_returns_error_message_if_roomname_not_string(self):
...
+		output = self.the_dojo.print_allocations("file1")
+		expected_output = "\nData has been successfully saved to file1.txt\n"
+		self.assertEqual(output, expected_output)
+
+	def test_print_allocations_if_filepath_exists(self):
+		self.the_dojo.print_allocations("file1")
+		self.assertTrue(os.path.isfile("file1.txt"))
+		os.remove("file1.txt")
+
+	def test_print_unallocated_with_filenamed_unspecified(self):
```
Comment (maquchizi) [inline test_cases/test_dojo.py]: *filename

**Review by maquchizi — APPROVED** [review]
Proceed to functionality.

**Pull request merged**